ui/de-lib/src/main/java/org/iplantc/de/diskResource/client/views/widgets/MultiFileSelectorField.java

```diff
@@ -130,23 +133,25 @@ public FileSelectDialogHideHandler(final FileSelectDialog dlg,
         }
 
         @Override
-        public void onHide(HideEvent event) {
-            Set<File> files = diskResourceUtil.filterFiles(dlg.getDiskResources());
-            if (files.isEmpty()) {
-                return;
-            }
-            store.addAll(files);
-            if (userSettings.isRememberLastPath() && store.size() > 0) {
-                userSettings.setLastPath(diskResourceUtil.parseParent(store.get(0).getPath()));
-                eventBus.fireEvent(new LastSelectedPathChangedEvent(true));
-            }
-            ValueChangeEvent.fire(MultiFileSelectorField.this,
-                                  Lists.<HasPath> newArrayList(store.getAll()));
-            setValue(Lists.<HasPath> newArrayList(store.getAll()));
+        public void onDialogHide(DialogHideEvent event) {
+        	if(event.getHideButton().equals(PredefinedButton.OK)) {
+	            Set<File> files = diskResourceUtil.filterFiles(dlg.getDiskResources());
+	            if (files.isEmpty()) {
+	                return;
+	            }
+	            store.addAll(files);
+	            if (userSettings.isRememberLastPath() && store.size() > 0) {
+	                userSettings.setLastPath(diskResourceUtil.parseParent(store.get(0).getPath()));
+	                eventBus.fireEvent(new LastSelectedPathChangedEvent(true));
+	            }
```

Comment:
Though, hm. Should the "remember last path" functionality still work even when the dialog is cancelled? With this change, it looks like the path will only be remembered if "OK" is clicked, which may or may not be the intent.